consensus/core/src/authority_service.rs

```diff
@@ -129,14 +129,40 @@ impl<C: CoreThreadDispatcher> NetworkService for AuthorityService<C> {
                 .metrics
                 .node_metrics
                 .block_timestamp_drift_wait_ms
-                .with_label_values(&[peer_hostname])
+                .with_label_values(&[peer_hostname, &"handle_send_block"])
                 .inc_by(forward_time_drift.as_millis() as u64);
             sleep(forward_time_drift).await;
         }
 
-        // Observe the block for the highest commit. When local commit is lagging too much,
+        // Observe the block for the commit votes. When local commit is lagging too much,
         // commit sync will be triggered.
-        self.highest_commit_monitor.observe(&verified_block);
+        self.commit_vote_monitor.observe(&verified_block);
+
+        // Reject blocks when local commit index is lagging too far from quorum commit index.
+        // Since the main issue of too many suspended blocks is memory usage not CPU,
+        // it is ok to reject after block verifications instead of before.
```

Comment:
is there any benefit to doing it after verification? seems like wasted effort, even if its not a bottleneck